### DISCUSSION TITLE
Confusion about what isGoogleBarcodeScannerModuleAvailable does and why installGoogleBarcodeScannerModule is needed

### DISCUSSION THREAD

**Discussion by basvdijk** [capawesome-team/capacitor-mlkit#84] [Q&A] [answered]
There is a function called `isGoogleBarcodeScannerModuleAvailable()`. According to the docs at https://capawesome.io/plugins/mlkit/barcode-scanning/#isgooglebarcodescannermoduleavailableresult

> Whether or not the Google [Barcode](https://capawesome.io/plugins/mlkit/barcode-scanning/#barcode) Scanner module is available.

For me it is unclear if "available" is the same as installed. Or that "available" refers to that the module is available to be installed.

Then there is the `installGoogleBarcodeScannerModule()` function. 

> Whether or not the Google [Barcode](https://capawesome.io/plugins/mlkit/barcode-scanning/#barcode) Scanner module is available.](https://capawesome.io/plugins/mlkit/barcode-scanning/#installgooglebarcodescannermodule)

For me it is unclear when I have to call this function and how I can determine if this module needs to be installed.

I am also wondering why the developer needs to take care of this and why the plugin can not install this module if it is not available.

**Comment by robingenz** [comment, marked as answer]
Thank you for your request. 

> For me it is unclear if "available" is the same as installed. Or that "available" refers to that the module is available to be installed.

Yes, `available` means `installed` in this case. I will improve the documentation.

> I am also wondering why the developer needs to take care of this and why the plugin can not install this module if it is not available.

The problem is that this installation can take some time (depending on the network connection etc.) and sometimes it fails completely (e.g. if there is no network connection). As a developer, I would like to have the ability to react to any events and possibly display appropriate instructions to the user. Before I show the user the button to scan a barcode I want to know if the module is available at all. If not, I start the installation and show the user a progress indicator.
However, I'm always open to ideas for improvement if you have any.

**Reply by basvdijk** [reply]
Thanks for the fast reply, this clarifies a lot!
